frontend/request.go

```diff
@@ -134,7 +135,40 @@ func marshalDockerfile(ctx context.Context, dt []byte, opts ...llb.ConstraintsOp
 	return st.Marshal(ctx)
 }
 
-func ForwardToSigner(ctx context.Context, client gwclient.Client, cfg *dalec.PackageSigner, s llb.State) (llb.State, error) {
+func MaybeSign(ctx context.Context, client gwclient.Client, st llb.State, spec *dalec.Spec, targetKey string) (llb.State, error) {
+	if signingDisabled(client) {
+		return st, nil
```

Comment:
As a follow-up, we should look at throwing a warning back to the client when signing is disabled.